src/components/PatientSearchbar.tsx

```diff
@@ -38,7 +38,7 @@ export function PatientSearchbar<T>({
     } else {
       setFilteredItems(data);
     }
-  }, [queryStr, setFilteredItems, village, data]);
+  }, [queryStr, setFilteredItems, village, data, filterFunction]);
```

Comment:
The filterFunction seems to be created on each render, resulting in useEffect constantly being ran.
```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [queryStr, setFilteredItems, village, data]);
```